### PR TITLE
Correct Room Service event timing

### PR DESCRIPTION
**Pull request by DaWoblefet**
Room Service currently uses onUpdate(), but it should use onStart() (same timing as Air Balloon), and some sort of event that is callable after a pseudoWeather is set (picked ``onAnyPseudoWeatherStart``).

### CODE REVIEW

File: sim/dex-conditions.ts [hunk 1]
```diff
@@ -375,6 +375,7 @@ export interface EventMethods {
 	onAnyNegateImmunity?: ((this: Battle, pokemon: Pokemon, type: string) => boolean | void) | boolean;
 	onAnyOverrideAction?: (this: Battle, pokemon: Pokemon, target: Pokemon, move: ActiveMove) => string | void;
 	onAnyPrepareHit?: CommonHandlers['ResultSourceMove'];
+	onAnyPseudoWeatherStart?: (this: Battle, target: Pokemon, source: Pokemon, pseudoWeather: Condition) => void;
```
[the remark this answers is not in the document]
Comment (KrisXV) [reply sim/dex-conditions.ts:378]: I believe it was just inserted alphabetically

File: data/items.ts [hunk 2]
```diff
@@ -4576,7 +4576,13 @@ export const Items: {[itemid: string]: ItemData} = {
 		fling: {
 			basePower: 100,
 		},
-		onUpdate(pokemon) {
+		onStart(pokemon) {
+			if (this.field.getPseudoWeather('trickroom')) {
```
Comment (Karthik99999) [inline data/items.ts:4580]:
```suggestion
			if (!pokemon.ignoringItem() && this.field.getPseudoWeather('trickroom')) {
```

File: sim/dex-conditions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (KrisXV) [reply sim/dex-conditions.ts:378]: Ah fair

[the remark this answers is not in the document]
Comment (DaWoblefet) [reply sim/dex-conditions.ts:378]: Yea, feel free to move it however you please in that PR.

**Pull request merged**